src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -715,6 +721,10 @@ def _push(
         # pylint: disable=protected-access
         if isinstance(other, (FormulaEngine, FormulaEngine3Phase)):
             self._steps.append((TokenType.COMPONENT_METRIC, other))
+        elif isinstance(
+            other, (Power, Current, Voltage, Energy, Frequency, Percentage)
+        ):
```

Comment:
I think using the base class is cleaner and probably requires less maintenance:

```suggestion
        elif isinstance(other, Quantity):
```